app/models/notification/purging.rb

```diff
@@ -0,0 +1,28 @@
+class Notification
+  module Purging
+    extend ActiveSupport::Concern
+    include PurgingMixin
+
+    module ClassMethods
+      def purge_date
+        nil
+      end
+
+      def purge_window_size
+        ::Settings.notifications.history.purge_window_size
+      end
+
+      # Collects the notifications that have expired based on the expires_in
+      # value for each notification type.  The "older_than" value is not used
+      # for this function, so it is named as `_`.
+      def purge_scope(_)
+        Notification.joins(:notification_type)
+                    .where("notifications.created_at + notification_types.expires_in * INTERVAL '1 second' < now()")
```

Comment:
this is very slick.

Did you try the math operations all on the right hand side?
it will be more index friendly. (not sure if any of this will ever hit indexes)